### PR TITLE
VHost limits

### PR DESCRIPTION
**Pull request by carlhoerberg**
https://trello.com/c/HfiVcZPH/100-add-support-for-vhost-limits

### CODE REVIEW

**Comment by carlhoerberg** [comment]
Screenshot 2022-08-12 at 11-50-13 _ LavinMQ

File: src/lavinmq/http/controller/vhost-limits.cr [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        put "/api/vhost-limits/:vhost/:type" do |context, params|
+          u = user(context)
+          refuse_unless_administrator(context, u)
+          context.response.status_code = 400
+          with_vhost(context, params) do |vhost|
+            v = @amqp_server.vhosts[vhost]
+            if body = context.request.body
+              json = JSON.parse(body)
+              if value = json["value"]?.try &.as_i?
+                value = nil if value < 0
```
Comment (jage) [inline src/lavinmq/http/controller/vhost-limits.cr:30]: Should we reject the request if it's "invalid", now we silently remove the limit.

Comment (carlhoerberg) [reply src/lavinmq/http/controller/vhost-limits.cr:30]: Only 204 on valid values, 400 for all other requests

Comment (jage) [reply src/lavinmq/http/controller/vhost-limits.cr:30]: Yeah, I was thinking we make `-1` invalid. Now it's valid, but setting nil.

Comment (carlhoerberg) [reply src/lavinmq/http/controller/vhost-limits.cr:30]: It's correct behavior, it's the way to delete the limit

Comment (carlhoerberg) [reply src/lavinmq/http/controller/vhost-limits.cr:30]: Same as for rabbitmq

**Pull request merged**